MergerLogic/DataTypes/HttpDataSource.cs

```diff
@@ -10,10 +9,20 @@ public abstract class HttpDataSource : Data<IHttpSourceUtils>
         protected IEnumerator<Tile[]> Batches;
         protected int BatchIndex = 0;
         protected HttpDataSource(IServiceProvider container,
-            DataType type, string path, int batchSize, Extent extent, GridOrigin origin, int maxZoom, int minZoom = 0, bool isOneXOne = false)
-            : base(container, type, path, batchSize, isOneXOne, origin)
+            DataType type, string path, int batchSize, Extent extent, GridOrigin? origin, Grid? grid, int maxZoom, int minZoom = 0)
+            : base(container, type, path, batchSize, grid, origin)
         {
-            this.GenTileRanges(extent, origin, minZoom, maxZoom);
+            this.GenTileRanges(extent, this.Origin, minZoom, maxZoom);
+        }
+
+        protected override GridOrigin DefaultOrigin()
```

Comment:
shouldn't be here as TMS and WMTS has different origin (idk what XYZ origin is)

Reply:
Fixed.